app/controllers/sessions_controller.rb

```diff
@@ -32,4 +25,29 @@ def destroy
     Current.user = nil
     redirect_to login_path
   end
+
+  private
+
+  if omniauth_authenticaton_ongoing?
+    request.env['omniauth.auth'].present?
+  end
+
+  def find_user_for_password_authentication
+    return if params[:email].blank? && params[:password].blank?
+
+    person = Person.find_by(email: params[:email].strip.downcase)
+
+    @user = person&.personable
+  end
+
+  def find_user_for_google_authentication
+    auth = request.env['omniauth.auth']
+    return if auth.blank?
+
+    @user = User.find_or_create_by!(uid: auth[:uid]) do |user|
```

Comment:
Here, I could also check for `Feature.enabled?(:registration)`... (see `#find_current_user_based_on_http_header`